octoprint_mrbeam/iobeam/lid_handler.py

```diff
@@ -297,50 +304,54 @@ def restart(self, pic_settings=None, cam_params=None, out_pic_size=None, blockin
-						replay=True))
-				self.stopEvent.clear()
-				if not self.stopEvent.wait(5.0):
-					self.work(recurse_nb=recurse_nb+1)
-			else:
-				self._logger.exception(" %s, %s : Recursive restart : too many times, displaying Error message." % (e.__class__.__name__, e),
+			if self.is_initial_calibration:
+				self.save_debug_images = True
+	
+			if not PICAMERA_AVAILABLE:
+				self._logger.warn("Camera disabled. Not all required modules could be loaded at startup. ")
+				self.stopEvent.set()
+				return
+	
+			self._logger.debug("Starting the camera now.")
+			try:
+				with MrbCamera(octoprint_mrbeam.camera.MrbPicWorker(maxSize=2, debug=self.debug),
+	                           # framerate=8,
+	                           resolution=octoprint_mrbeam.camera.LEGACY_STILL_RES,  # TODO camera.DEFAULT_STILL_RES,
+	                           stopEvent=self.stopEvent,) as cam:
+					self.serve_pictures(cam, pic_settings=pic_settings, cam_params=cam_params, out_pic_size=out_pic_size)
+				if recurse_nb > 0:
+					self._logger.info("Camera recovered")
+					self._analytics_handler.add_camera_session_details(exc.msgForAnalytics(exc.CAM_CONNRECOVER))
+			except exc.CameraConnectionException as e:
+				self._logger.warning(" %s, %s : %s" % (e.__class__.__name__, e, exc.msg(exc.CAM_CONN)),
 				                       analytics=exc.CAM_CONN)
-				self._plugin.user_notification_system.show_notifications(
-					self._plugin.user_notification_system.get_notification(
-						notification_id='err_cam_conn_err',
-						replay=True))
-			return
-		except Exception as e:
-			if e.__class__.__name__.startswith('PiCamera'):
-				self._logger.exception("PiCamera_Error_while_preparing_camera_%s_%s", e.__class__.__name__, e)
-			else:
-				self._logger.exception("Exception_while_preparing_camera_%s_%s", e.__class__.__name__, e)
-		self.stopEvent.set()
+				if recurse_nb < MAX_PIC_THREAD_RETRIES:
+					self._logger.info("Restarting work() after some sleep")
+					self._plugin.user_notification_system.show_notifications(
+						self._plugin.user_notification_system.get_notification(
+							notification_id='warn_cam_conn_err',
+							replay=True))
+					self.stopEvent.clear()
+					if not self.stopEvent.wait(5.0):
+						self.work(recurse_nb=recurse_nb+1)
+				else:
+					self._logger.exception(" %s, %s : Recursive restart : too many times, displaying Error message." % (e.__class__.__name__, e),
+					                       analytics=exc.CAM_CONN)
+					self._plugin.user_notification_system.show_notifications(
+						self._plugin.user_notification_system.get_notification(
+							notification_id='err_cam_conn_err',
+							replay=True))
+				return
+			except Exception as e:
+				if e.__class__.__name__.startswith('PiCamera'):
+					self._logger.exception("PiCamera_Error_while_preparing_camera_%s_%s", e.__class__.__name__, e)
+				else:
+					self._logger.exception("Exception_while_preparing_camera_%s_%s", e.__class__.__name__, e)
+			self.stopEvent.set()
+		except:
```

Comment:
I don't understand what this extra try/except achieves. There is already one inside and this one doesn't log the exception.